### ISSUE TITLE
NgOnInit called with disabled feature

### ISSUE THREAD

**Issue by felpel** [willmendesneto/ngx-feature-toggle#184]
**I'm submitting a ...** 
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
```

**Current behavior**

We're using a feature toggle to show or hide a section in a form. We're implementing `OnInit` to add the form group in its parent one. 

**Expected behavior**

We would expect that the component's `ngOnInit` method would not get called at all like it would happen if we used instead a `ngIf` directive. 

**Reproduction of the problem**

[Stackblitz demo](https://stackblitz.com/edit/angular-dmxzys)

**What is the motivation / use case for changing the behavior?**

The component is not rendered as expected, but nothing related to an inner component life cycle should be called at all.

**Please tell us about your environment:**

- OS: Windows 10;
- IDE: Visual Studio Code;
- Package manager: npm;
- HTTP Server: Webpack (`ng serve`);

* **Browser:** Tested on all major browsers available on Windows: Chrome, Firefox, Edge (+ Opera)

* **Language:** Typescript

* **Node (if applicable):** Probably not applicable, but `node --version = 8.11.1`

**Comment by willmendesneto** [comment]
Hi @felpel, thanks for open the issue and add the Stackblitz scenario. I'll take a look in the issue during this week.

**Comment by felpel** [comment]
By any chance, have you investigated a little bit about this issue @willmendesneto ?

**Comment by willmendesneto** [comment]
@felpel no luck, mate. Still investigating and digging to find a fix. If you have any ideas let's chat or you can open a pull request.

One interesting point is that I was able to simulate the issue without `ngx-feature-toggle` included, as you can see [in this Stackblitz link](https://stackblitz.com/edit/angular-huxftm?file=src%2Fapp%2Fapp.component.html).

**Comment by felpel** [comment]
I see, it seems that `ngOnInit` is still called even when the condition is hardcoded to `false` like:
```html
<div *ngIf="false">
  <hello name="Another Oups"></hello>
</div>
```

Should we use `onInit` or consider one of the other Angular lifecylcle methods?

**Comment by willmendesneto** [comment]
@felpel For now in your scenario, I encourage you to add your validations after `ngOnInit`. As you can see [here in this link](https://stackblitz.com/edit/angular-csbfwb?file=src/app/hello.component.ts) you can use it on `ngDoCkeck` or any lifecycle event after ngOnInit, as described in the Angular Docs.

https://angular.io/guide/lifecycle-hooks

**Comment by Andrey-Pavlov** [comment]
@willmendesneto My observations. Now it works with *ngIf='false' correctly but doesn't work when the component inside the code.
```
<feature-toggle [featureName]="obj.flag" showWhenDisabled>
    <my-cmp></my-cmp>
</feature-toggle>
```
I make http calls inside ngOnInit and now I have x2 http requests

**Comment by willmendesneto** [comment]
@Andrey-Pavlov yeah, I checked the new version and the issue still persists. You can check this comment and try the stackblitz example sharing that. 

https://github.com/willmendesneto/ngx-feature-toggle/issues/184#issuecomment-426477601

**Comment by Andrey-Pavlov** [comment]
> @Andrey-Pavlov yeah, I checked the new version and the issue still persists. You can check this comment and try the stackblitz example sharing that.
> 
> [#184 (comment)](https://github.com/willmendesneto/ngx-feature-toggle/issues/184#issuecomment-426477601)

@willmendesneto  Ok, I see the point now. Thank you. May be you know any valid workarounds to preserve http calls in ngOnInit?

**Comment by willmendesneto** [comment]
@Andrey-Pavlov @felpel I'm landing a pull request with this fix. Unfortunately, that will be a breaking change. After discussion with other GDE's we just realized that the proper way is using directive (even knowing that the `ng-template` logic can work). So I'm adding 2 directives and having the same features we already have in place.

More details in this pull request https://github.com/willmendesneto/ngx-feature-toggle/pull/186

**Comment by willmendesneto** [comment]
@felpel @Andrey-Pavlov I published a new version of `ngx-feature-toggle`. This fix was added in the `v7.0.0`, please make sure you updated your apps.

Closing this issue right now. Thank you so much for finding this issue, folks!
